geofileops/util/_geoops_sql.py

```diff
@@ -1442,27 +1442,34 @@ def export_by_location(
     input1_layer_rtree = "rtree_{input1_layer}_{input1_geometrycolumn}"
     input2_layer_rtree = "rtree_{input2_layer}_{input2_geometrycolumn}"
 
-    # Prepare spatial relation column and filter
-    (
-        spatial_relation_column,
-        spatial_relation_filter,
-        true_for_disjoint,
-    ) = _prepare_filter_by_location_fields(spatial_relations_query)
-
     # Subdivide the 2nd layer if applicable to speed up further processing.
     tmp_dir = _io_util.create_tempdir(f"geofileops/{operation_name}")
     input_to_compare_with_subdivided_path = _subdivide_layer(
         path=input_to_compare_with_path,
         layer=input_to_compare_with_layer,
         output_path=tmp_dir / "subdivided/input_to_compare_with_layer.gpkg",
         subdivide_coords=subdivide_coords,
+        keep_fid=True if subdivide_coords > 0 else False,
         nb_parallel=nb_parallel,
         batchsize=batchsize,
         operation_prefix=f"{operation_name}/",
     )
     if input_to_compare_with_subdivided_path is not None:
         input_to_compare_with_path = input_to_compare_with_subdivided_path
 
+    # Prepare spatial relation column and filter
+    (
+        spatial_relation_column,
+        spatial_relation_filter,
+        true_for_disjoint,
+        groupby,
+    ) = _prepare_filter_by_location_fields(
+        query=spatial_relations_query,
+        subdivide_coords=subdivide_coords
+        if input_to_compare_with_subdivided_path is not None
+        else 0,
+    )
+
     # Different query if intersecting features need to be unioned...
     if (
         true_for_disjoint is False
```

Comment:
```suggestion
```
It looks like this is an obsolete line of code, as I suppose this makes the `if true_for_disjoint:` later on unreachable? However, if I remove it, a few tests start to fail?